raiden/network/proxies/netting_channel.py

```diff
@@ -425,15 +425,13 @@ def settle(self):
                 node=pex(self.node_address),
             )
 
-        if not self.channel_operations_lock.acquire(0):
+        if not self.channel_operations_lock.acquire(blocking=False):
             raise ChannelBusyError(
                 f'Channel with address {self.address} is '
                 f'busy with another ongoing operation'
             )
 
-        self.channel_operations_lock.release()
-
-        with self.channel_operations_lock:
+        else:
```

Comment:
I also considered this approach, but the issue is the lock won't be released if any of the logic after it throws an exception. The alternative was to wrap the block in a try-finally and to release the lock in the finally block.

Reply:
Good point. I added a context manager to release the locks, this should handle it correctly now.